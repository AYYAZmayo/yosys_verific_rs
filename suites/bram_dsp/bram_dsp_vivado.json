{
    "description": "Only LUT benchmarks where LUT > 100",
    "tool": "vivado",
    "yosys":
        {
            "yosys_path": "yosys/install/bin/yosys",
            "yosys_template_script": "scripts/synth/yosys/yosys_template_synth_rs_optional.ys",
            "verific": true,
            "synth_rs" :
                {
                    "-tech": "genesis",
                    "-goal": "area",
                    "-de": true,
                    "-verilog" : "synthesized.v"
                }
        },
    "vivado": 
        {
            "vivado_template_script": "scripts/synth/vivado/vivado_v1_template.tcl"
        },
    "diamond":
        {
            "diamond_template_script": "scripts/synth/diamond/diamond_template.tcl"
        },
    "num_process": 8,
    "timeout": 21600,
    "benchmarks": [
        {
            "name": "bitonic_mesh",
            "rtl_path": "RTL_Benchmark/Verilog/ql_design/bitonic_mesh/rtl",
            "top_module": "system100"
        },
        {
            "name": "cf_fft_256_8",
            "rtl_path": "RTL_Benchmark/Verilog/ql_design/cf_fft_256_8/rtl",
            "top_module": "cf_fft_256_8"
        },
        {
            "name": "des90",
            "rtl_path": "RTL_Benchmark/Verilog/ql_design/des90/rtl", 
            "top_module": "system90"
        },
        {
            "name": "stereovision2",
            "rtl_path": "RTL_Benchmark/Verilog/ql_design/stereovision2/rtl",
            "top_module": "sv_chip2_hierarchy_no_mem"
        }
    ]
}
