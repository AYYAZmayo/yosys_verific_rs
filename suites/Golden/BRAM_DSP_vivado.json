{
    "description": "Golden suite of benchmarks to run with Vivado.",
    "tool": "vivado",
    "vivado":
        {
            "vivado_template_script": "scripts/synth/vivado/vivado_bram_dsp_template.tcl"
        },
    "num_process": 8,
    "timeout": 21600,
    "deactivated_benchmarks": [
        
    ],
    "benchmarks": [
        {
            "name" : "EDA-402-hameed",
            "rtl_path": "RTL_Benchmark/Verilog/yosys_validation/EDA-402/EDA-402-hameed/",
            "top_module": "rams_sp_reg_addr_1024x32"
        },
        {
            "name" : "EDA-629-1",
            "rtl_path": "RTL_Benchmark/Verilog/yosys_validation/EDA-629/EDA-629-1",
            "top_module": "ram_simple_dp_sync_read_4096x32"
        },
        {
            "name" : "EDA-629-2",
            "rtl_path": "RTL_Benchmark/Verilog/yosys_validation/EDA-629/EDA-629-2",
            "top_module": "ram_simple_dp_sync_reg_read_4096x32"
        },
        {
            "name" : "EDA-713",
            "rtl_path": "RTL_Benchmark/Verilog/yosys_validation/EDA-713",
            "top_module": "rams_sp_wf_rst_en_1024x16"
        },
        {    "name" : "SDP_RF",
            "rtl_path": "RTL_Benchmark/Verilog/yosys_validation/SDP_RF/",
            "top_module": "ram_simple_dp_synch_rf_1024x32"
        },
        {    "name" : "SP_WF",
            "rtl_path": "RTL_Benchmark/Verilog/yosys_validation/SP_WF/",
            "top_module": "rams_sp_wf_1024x7"
        },
        {
            "name" : "EDA-860",
            "rtl_path": "RTL_Benchmark/Verilog/yosys_validation/EDA-860/",
            "top_module": "ram_true_dp_dc_512x32"
        },
        {
            "name" : "EDA-864",
            "rtl_path": "RTL_Benchmark/Verilog/yosys_validation/EDA-864/",
            "top_module": "rams_sp_re_we_rf_1024x32"
        },
        {
            "name" : "EDA-869",
            "rtl_path": "RTL_Benchmark/Verilog/yosys_validation/EDA-869/",
            "top_module": "ram_true_dp_dc_4096x36"
        },
        {
            "name" : "EDA-879",
            "rtl_path": "RTL_Benchmark/Verilog/yosys_validation/EDA-879/",
            "top_module": "rams_sp_wf_rst_en_1024x16"
        },
        {
            "name" : "EDA-880",
            "rtl_path": "RTL_Benchmark/Verilog/yosys_validation/EDA-880/",
            "top_module": "tdp_256x8"
        },
        {
            "name" : "EDA-881",
            "rtl_path": "RTL_Benchmark/Verilog/yosys_validation/EDA-881/",
            "top_module": "rams_sp_reg_addr_1024x36"
        },
        {
            "name" : "EDA-884",
            "rtl_path": "RTL_Benchmark/Verilog/yosys_validation/EDA-884/",
            "top_module": "ram_true_reg_addr_dp_1024x32"
        },
        {
            "name" : "EDA-885",
            "rtl_path": "RTL_Benchmark/Verilog/yosys_validation/EDA-885/",
            "top_module": "ram_true_reg_addr_dp_1024x8"
        },
        {
            "name" : "EDA-887",
            "rtl_path": "RTL_Benchmark/Verilog/yosys_validation/EDA-887/",
            "top_module": "ram_true_dp_wf_1024x32"
        },
        {
            "name" : "EDA-890",
            "rtl_path": "RTL_Benchmark/Verilog/yosys_validation/EDA-890/",
            "top_module": "rams_sp_re_prio_we_rst_1024x32"
        },
        {
            "name" : "EDA-891",
            "rtl_path": "RTL_Benchmark/Verilog/yosys_validation/EDA-891/",
            "top_module": "ram_simple_dp_sync_reg_read_1024x64"
        } ,
        {
            "name" : "EDA-1090_512x8",
            "rtl_path": "RTL_Benchmark/Verilog/yosys_validation/EDA-1090/512x8",
            "top_module": "rams_sp_reg_addr_readmem_512x8"
        },
        {
            "name" : "EDA-1090_1024x1",
            "rtl_path": "RTL_Benchmark/Verilog/yosys_validation/EDA-1090/1024x1",
            "top_module": "rams_sp_reg_addr_readmem_1024x1"
        },
        {
            "name" : "EDA-1091",
            "rtl_path": "RTL_Benchmark/Verilog/yosys_validation/EDA-1091/",
            "top_module": "aes_256_sr_top"
        },
        {
            "name" : "EDA-871_1024x4",
            "rtl_path": "RTL_Benchmark/Verilog/yosys_validation/EDA-871/bytewrite_sdp_ram_nc_1024x4",
            "top_module": "bytewrite_sdp_ram_nc_1024x4"
        },
        {
            "name" : "EDA-871_1024x8",
            "rtl_path": "RTL_Benchmark/Verilog/yosys_validation/EDA-871/bytewrite_sdp_ram_nc_1024x8",
            "top_module": "bytewrite_sdp_ram_nc_1024x8"
        },
        {
            "name" : "EDA-871_1024x12",
            "rtl_path": "RTL_Benchmark/Verilog/yosys_validation/EDA-871/bytewrite_sdp_ram_nc_1024x12",
            "top_module": "bytewrite_sdp_ram_nc_1024x12"
        },
        {
            "name" : "EDA-871_1024x16",
            "rtl_path": "RTL_Benchmark/Verilog/yosys_validation/EDA-871/bytewrite_sdp_ram_nc_1024x16",
            "top_module": "bytewrite_sdp_ram_nc_1024x16"
        },
        {
            "name" : "EDA-871_1024x20",
            "rtl_path": "RTL_Benchmark/Verilog/yosys_validation/EDA-871/bytewrite_sdp_ram_nc_1024x20",
            "top_module": "bytewrite_sdp_ram_nc_1024x20"
        },
        {
            "name" : "EDA-871_1024x24",
            "rtl_path": "RTL_Benchmark/Verilog/yosys_validation/EDA-871/bytewrite_sdp_ram_nc_1024x24",
            "top_module": "bytewrite_sdp_ram_nc_1024x24"
		},
        {
            "name" : "EDA-871_1024x28",
            "rtl_path": "RTL_Benchmark/Verilog/yosys_validation/EDA-871/bytewrite_sdp_ram_nc_1024x28",
            "top_module": "bytewrite_sdp_ram_nc_1024x28"
        },
        {
            "name" : "EDA-871_1024x32",
            "rtl_path": "RTL_Benchmark/Verilog/yosys_validation/EDA-871/bytewrite_sdp_ram_nc_1024x32",
            "top_module": "bytewrite_sdp_ram_nc_1024x32"
        },
        {
            "name" : "EDA-893",
            "rtl_path": "RTL_Benchmark/Verilog/yosys_validation/EDA-893/",
            "top_module": "bytewrite_sp_ram_wf"
        },
        {
            "name" : "EDA-670",
            "rtl_path": "RTL_Benchmark/Verilog/yosys_validation/EDA-670/",
            "top_module": "bytewrite_sdp_ram_rf"
        },
        {
            "name" : "EDA-863",
            "rtl_path": "RTL_Benchmark/Verilog/yosys_validation/EDA-863",
            "top_module": "asym_ram_sdp_wide_sync_read"
        },
        {
            "name" : "EDA-896",
            "rtl_path": "RTL_Benchmark/Verilog/yosys_validation/EDA-896",
            "top_module": "asym_ram_sdp_wide_write"
        },
        {
            "name" : "EDA-899",
            "rtl_path": "RTL_Benchmark/Verilog/yosys_validation/EDA-899",
            "top_module": "asym_ram_sdp_read_wider_dc"
        },
        {
            "name" : "EDA-1046",
            "rtl_path": "RTL_Benchmark/Verilog/yosys_validation/EDA-1046",
            "top_module": "asym_ram_sdp_write_wider"
        },
        {
            "name" : "EDA-720",
            "rtl_path": "RTL_Benchmark/Verilog/yosys_validation/EDA-720",
            "top_module": "asym_ram_sdp_wide_sync_read_via_part_selection"
        },
        {
            "name" : "EDA-1318",
            "rtl_path": "RTL_Benchmark/Verilog/yosys_validation/EDA-1318",
            "top_module": "two_sdp"
        },
        {
            "name" : "EDA-1281_8192x16",
            "rtl_path": "RTL_Benchmark/Verilog/yosys_validation/EDA-1281/8192x16",
            "top_module": "ram_true_dp_dc_8192x16"
        },
        {
            "name" : "EDA-1281_8192x18",
            "rtl_path": "RTL_Benchmark/Verilog/yosys_validation/EDA-1281/8192x18",
            "top_module": "ram_true_dp_dc_8192x18"
        },
        {
            "name" : "EDA-1278",
            "rtl_path": "RTL_Benchmark/Verilog/yosys_validation/EDA-1278",
            "top_module": "ram_true_dp_out_reg_1024x32"
        },
        {
            "name" : "EDA-1459",
            "rtl_path": "RTL_Benchmark/Verilog/yosys_validation/EDA-1459",
            "top_module": "rams_sp_re_prio_we_rst_512x16_block"
        },
        {
            "name" : "EDA-1436",
            "rtl_path": "RTL_Benchmark/Verilog/yosys_validation/EDA-1436",
            "top_module": "rams_sp_rf_rst_async_512x16_block"
        },
        {   
            "name" : "accum_output_shifted_rounded", 
            "rtl_path": "RTL_Benchmark/Verilog/yosys_validation/DSP_Designs/accum_output_shifted_rounded",
            "top_module": "accum_output_shifted_rounded"
        },
        {   
            "name" : "accum_output_shifted_saturated", 
            "rtl_path": "RTL_Benchmark/Verilog/yosys_validation/DSP_Designs/accum_output_shifted_saturated",
            "top_module": "accum_output_shifted_saturated"
        },
        {   
            "name" : "accumulator", 
            "rtl_path": "RTL_Benchmark/Verilog/yosys_validation/DSP_Designs/accumulator",
            "top_module": "accumulator"
        },
        {   
            "name" : "add_output_of_four_multipliers", 
            "rtl_path": "RTL_Benchmark/Verilog/yosys_validation/DSP_Designs/add_output_of_four_multipliers",
            "top_module": "add_output_of_four_multipliers"
        },
        {   
            "name" : "add_shifted_input_to_the_mul_output", 
            "rtl_path": "RTL_Benchmark/Verilog/yosys_validation/DSP_Designs/add_shifted_input_to_the_mul_output",
            "top_module": "add_shifted_input_to_the_mul_output"
        },
        {   
            "name" : "cic_decimator", 
            "rtl_path": "RTL_Benchmark/Verilog/yosys_validation/DSP_Designs/cic_decimator",
            "top_module": "cic_decimator"
        },
        {   
            "name" : "complex_multiplier", 
            "rtl_path": "RTL_Benchmark/Verilog/yosys_validation/DSP_Designs/complex_multiplier",
            "top_module": "complex_multiplier"
        },
        {   
            "name" : "dct", 
            "rtl_path": "RTL_Benchmark/Verilog/yosys_validation/DSP_Designs/dct",
            "top_module": "dct"
        },
        {   
            "name" : "dsp_add_mul_output_to_accum_20lsb", 
            "rtl_path": "RTL_Benchmark/Verilog/yosys_validation/DSP_Designs/dsp_add_mul_output_to_accum_20lsb",
            "top_module": "dsp_add_mul_output_to_accum_20lsb"
        },
        {   
            "name" : "dsp_add_shifted_input_to_the_mul_coeff0_output", 
            "rtl_path": "RTL_Benchmark/Verilog/yosys_validation/DSP_Designs/dsp_add_shifted_input_to_the_mul_coeff0_output",
            "top_module": "dsp_add_shifted_input_to_the_mul_coeff0_output"
        },
        {   
            "name" : "dsp_eight_mult", 
            "rtl_path": "RTL_Benchmark/Verilog/yosys_validation/DSP_Designs/dsp_eight_mult",
            "top_module": "dsp_eight_mult"
        },
        {   
            "name" : "dsp_fractured_accum_output_shifted_rounded", 
            "rtl_path": "RTL_Benchmark/Verilog/yosys_validation/DSP_Designs/dsp_fractured_accum_output_shifted_rounded",
            "top_module": "dsp_fractured_accum_output_shifted_rounded"
        },
        {   
            "name" : "dsp_fractured_signed_mul_comb", 
            "rtl_path": "RTL_Benchmark/Verilog/yosys_validation/DSP_Designs/dsp_fractured_signed_mul_comb",
            "top_module": "dsp_fractured_signed_mul_comb"
        },
        {   
            "name" : "dsp_mul_complex_eq", 
            "rtl_path": "RTL_Benchmark/Verilog/yosys_validation/DSP_Designs/dsp_mul_complex_eq",
            "top_module": "dsp_mul_complex_eq"
        },
        {   
            "name" : "dsp_mul_parameterized", 
            "rtl_path": "RTL_Benchmark/Verilog/yosys_validation/DSP_Designs/dsp_mul_parameterized",
            "top_module": "dsp_mul_parameterized"
        },
        {   
            "name" : "dsp_mul_signed_accum_add_only", 
            "rtl_path": "RTL_Benchmark/Verilog/yosys_validation/DSP_Designs/dsp_mul_signed_accum_add_only",
            "top_module": "dsp_mul_signed_accum_add_only"
        },
        {   
            "name" : "dsp_mul_signed_comb", 
            "rtl_path": "RTL_Benchmark/Verilog/yosys_validation/DSP_Designs/dsp_mul_signed_comb",
            "top_module": "dsp_mul_signed_comb"
        },
        {   
            "name" : "dsp_mul_signed_in_not_reg_out_is_reg", 
            "rtl_path": "RTL_Benchmark/Verilog/yosys_validation/DSP_Designs/dsp_mul_signed_in_not_reg_out_is_reg",
            "top_module": "dsp_mul_signed_in_not_reg_out_is_reg"
        },
        {   
            "name" : "dsp_mul_signed_in_not_reg_with_accum_output_is_not_reg", 
            "rtl_path": "RTL_Benchmark/Verilog/yosys_validation/DSP_Designs/dsp_mul_signed_in_not_reg_with_accum_output_is_not_reg",
            "top_module": "dsp_mul_signed_in_not_reg_with_accum_output_is_not_reg"
        },
        {   
            "name" : "dsp_mul_signed_reg", 
            "rtl_path": "RTL_Benchmark/Verilog/yosys_validation/DSP_Designs/dsp_mul_signed_reg",
            "top_module": "dsp_mul_signed_reg"
        },
        {   
            "name" : "dsp_mul_signed_reg_active_low_async_reset", 
            "rtl_path": "RTL_Benchmark/Verilog/yosys_validation/DSP_Designs/dsp_mul_signed_reg_active_low_async_reset",
            "top_module": "dsp_mul_signed_reg_active_low_async_reset"
        },
        {   
            "name" : "dsp_mul_signed_reg_active_low_sync_reset", 
            "rtl_path": "RTL_Benchmark/Verilog/yosys_validation/DSP_Designs/dsp_mul_signed_reg_active_low_sync_reset",
            "top_module": "dsp_mul_signed_reg_active_low_sync_reset"
        },
        {   
            "name" : "dsp_mul_signed_reg_output_is_not_reg", 
            "rtl_path": "RTL_Benchmark/Verilog/yosys_validation/DSP_Designs/dsp_mul_signed_reg_output_is_not_reg",
            "top_module": "dsp_mul_signed_reg_output_is_not_reg"
        },
        {   
            "name" : "dsp_mul_signed_reg_with_accum", 
            "rtl_path": "RTL_Benchmark/Verilog/yosys_validation/DSP_Designs/dsp_mul_signed_reg_with_accum",
            "top_module": "dsp_mul_signed_reg_with_accum"
        },
        {   
            "name" : "dsp_mul_signed_reg_with_accum_output_is_reg", 
            "rtl_path": "RTL_Benchmark/Verilog/yosys_validation/DSP_Designs/dsp_mul_signed_reg_with_accum_output_is_reg",
            "top_module": "dsp_mul_signed_reg_with_accum_output_is_reg"
        },
        {   
            "name" : "dsp_mul_unsigned_comb", 
            "rtl_path": "RTL_Benchmark/Verilog/yosys_validation/DSP_Designs/dsp_mul_unsigned_comb",
            "top_module": "dsp_mul_unsigned_comb"
        },
        {   
            "name" : "dsp_mul_unsigned_parameterized", 
            "rtl_path": "RTL_Benchmark/Verilog/yosys_validation/DSP_Designs/dsp_mul_unsigned_parameterized",
            "top_module": "dsp_mul_unsigned_parameterized"
        },
        {   
            "name" : "dsp_mul_unsigned_reg", 
            "rtl_path": "RTL_Benchmark/Verilog/yosys_validation/DSP_Designs/dsp_mul_unsigned_reg",
            "top_module": "dsp_mul_unsigned_reg"
        },
        {   
            "name" : "dsp_multiplier_accum_with_add", 
            "rtl_path": "RTL_Benchmark/Verilog/yosys_validation/DSP_Designs/dsp_multiplier_accum_with_add",
            "top_module": "dsp_multiplier_accum_with_add"
        },
        {   
            "name" : "dsp_multiplier_accum_with_add_and_sub", 
            "rtl_path": "RTL_Benchmark/Verilog/yosys_validation/DSP_Designs/dsp_multiplier_accum_with_add_and_sub",
            "top_module": "dsp_multiplier_accum_with_add_and_sub"
        },
        {   
            "name" : "dsp_z_o_wrt_out_select_i", 
            "rtl_path": "RTL_Benchmark/Verilog/yosys_validation/DSP_Designs/dsp_z_o_wrt_out_select_i",
            "top_module": "dsp_z_o_wrt_out_select_i"
        },
        {   
            "name" : "input_to_adder_and_mul_A_input_wrt_feedback_i", 
            "rtl_path": "RTL_Benchmark/Verilog/yosys_validation/DSP_Designs/input_to_adder_and_mul_A_input_wrt_feedback_i",
            "top_module": "input_to_adder_and_mul_A_input_wrt_feedback_i"
        },
        {   
            "name" : "instantiate_adder_in_mult_and_shifted_input_design", 
            "rtl_path": "RTL_Benchmark/Verilog/yosys_validation/DSP_Designs/instantiate_adder_in_mult_and_shifted_input_design",
            "top_module": "instantiate_adder_in_mult_and_shifted_input_design"
        },
        {   
            "name" : "instantiate_mul_in_accum", 
            "rtl_path": "RTL_Benchmark/Verilog/yosys_validation/DSP_Designs/instantiate_mul_in_accum",
            "top_module": "instantiate_mul_in_accum"
        },
        {   
            "name" : "load_accum", 
            "rtl_path": "RTL_Benchmark/Verilog/yosys_validation/DSP_Designs/load_accum",
            "top_module": "load_accum"
        },
        {   
            "name" : "matrix_mult_3x3", 
            "rtl_path": "RTL_Benchmark/Verilog/yosys_validation/DSP_Designs/matrix_mult_3x3",
            "top_module": "matrix_mult_3x3"
        },
        {   
            "name" : "mul_and_reflect_input_B_as_registered_out", 
            "rtl_path": "RTL_Benchmark/Verilog/yosys_validation/DSP_Designs/mul_and_reflect_input_B_as_registered_out",
            "top_module": "mul_and_reflect_input_B_as_registered_out"
        },
        {   
            "name" : "multiplier_adder_wrt_Reg_input_i", 
            "rtl_path": "RTL_Benchmark/Verilog/yosys_validation/DSP_Designs/multiplier_adder_wrt_Reg_input_i",
            "top_module": "multiplier_adder_wrt_Reg_input_i"
        },
        {   
            "name" : "reg_and_not_reg_input_mul_with_accum", 
            "rtl_path": "RTL_Benchmark/Verilog/yosys_validation/DSP_Designs/reg_and_not_reg_input_mul_with_accum",
            "top_module": "reg_and_not_reg_input_mul_with_accum"
        },
        {   
            "name" : "simple_multiplier_with_adder", 
            "rtl_path": "RTL_Benchmark/Verilog/yosys_validation/DSP_Designs/simple_multiplier_with_adder",
            "top_module": "simple_multiplier_with_adder"
        },
        {   
            "name" : "simple_unsigned_4tap_fir", 
            "rtl_path": "RTL_Benchmark/Verilog/yosys_validation/DSP_Designs/simple_unsigned_4tap_fir",
            "top_module": "simple_unsigned_4tap_fir"
        },
        {   
            "name" : "sixteen_mult_accum", 
            "rtl_path": "RTL_Benchmark/Verilog/yosys_validation/DSP_Designs/sixteen_mult_accum",
            "top_module": "sixteen_mult_accum"
        }
    ]
}
